src/Components/PaperEditor/ProgrammeScript/index.js

```diff
@@ -1,238 +1,270 @@
-      ]
-    };
-  }
 
-  updateVideoContextWidth = () => {
-    this.setState({
-      width: this.card.offsetWidth - 10
-    });
-  };
+import PreviewCanvas from '@bbc/digital-paper-edit-storybook/PreviewCanvas';
+import ProgrammeElements from '@bbc/digital-paper-edit-storybook/ProgrammeElements';
 
-  getTranscript = transcriptId => {
-    return this.props.transcripts.find(tr => tr.id === transcriptId);
-  };
+import Collection from '../../Firebase/Collection';
+import { withAuthorization } from '../../Session';
 
-  getPlayList = () => {
-    let startTime = 0;
-
-    return this.state.programmeScript.elements
-      .filter(element => element.type === 'paper-cut')
-      .map(element => {
-        // TODO: handle audio only type (eg for radio), get from transcript json?
-        const transcript = this.getTranscript(element.transcriptId);
-        const result = {
-          type: 'video',
-          start: startTime,
-          sourceStart: element.start,
-          duration: element.end - element.start,
-          src: transcript.url
-        };
+import ExportDropdown from './ExportDropdown/index';
+import ElementsDropdown from './ElementsDropdown/index';
+import getDataFromUserWordsSelection from './get-data-from-user-selection';
+import { divideWordsSelectionsIntoParagraphs, isOneParagraph } from './divide-words-selections-into-paragraphs';
 
-        startTime += result.duration;
+import PropTypes from 'prop-types';
 
-        return result;
+const ProgrammeScript = props => {
+  const transcripts = props.transcripts;
+  const papereditsId = props.match.params.papereditId;
+  const projectId = props.match.params.projectId;
+
+  const [ elements, setElements ] = useState();
+  const [ title, setTitle ] = useState('');
+  const [ resetPreview, setResetPreview ] = useState(false);
+
+  const SortableList = SortableContainer(({ children }) =>
+    <ul style={ { listStyle: 'none', padding: '0px' } }>
+      {children}
+    </ul>
+  );
+  
+  // Video Context Preview
+  const [ width, setWidth ] = useState(150);
+  const [ playlist, setPlaylist ] = useState();
+  const previewCardRef = useRef(null);
+
+  const PaperEditsCollection = new Collection(
+    props.firebase,
+    `/projects/${ projectId }/paperedits`
+  );
+
+  const handleSaveProgrammeScript = async () => {
+    console.log('Saving...');
+    if (elements) {
+      const newElements = JSON.parse(JSON.stringify(elements));
+      // finding an removing insert point before saving to server
+      // find insert point in list,
+      const insertPointElement = newElements.find(el => {
+        return el.type === 'insert';
       });
+      if (insertPointElement) {
+        // get insertpoint index
+        const indexOfInsertPoint = newElements.indexOf(insertPointElement);
+        newElements.splice(indexOfInsertPoint, 1);
+      }
+
+      // Not sure if this is right - probably need to add back a bunch of metadata.
```

Comment:
This comment is wrong - this doesn't overwrite data in Firestore. Unless the schema is wrong, this comment can be deleted.